okhttp/src/main/java/io/grpc/transport/okhttp/AsyncFrameWriter.java

```diff
@@ -228,6 +237,9 @@ public void run() {
     @Override
     public final void run() {
       try {
+        if (frameWriter == null) {
+          throw new RuntimeException("Has write operation before real frameWriter is initialized.");
```

Comment:
This description isn't quite true. It is describing one of two possible cases. The other case is that starting the connection failed and threw an exception. How about something that assumes a little less about the reason:

> Unable to perform write due to unavailable frameWriter

It would also seem reasonable to make it an IOException, since then it will fall into the normal logic of performing a write after the connection has shut down.